statsmodels/stats/nonparametric.py

```diff
@@ -662,3 +663,273 @@ def cohensd2problarger(d):
+        n_2 : int
+            Number of observations in the second sample.
+        overall_ranks : ndarray
+            Ranks of the pooled sample.
+        overall_ranks_1 : ndarray
+            Ranks of the first sample in the pooled sample.
+        overall_ranks_2 : ndarray
+            Ranks of the second sample in the pooled sample.
+        within_group_ranks_1 : ndarray
+            Internal ranks of the first sample.
+        within_group_ranks_2 : ndarray
+            Internal ranks of the second sample.
+        placements_1 : ndarray
+            Placements of the first sample in the pooled sample.
+        placements_2 : ndarray
+            Placements of the second sample in the pooled sample.
+
+    Notes
+    -----
+    * The overall rank for each observation is determined
+    by ranking all data points from both samples combined
+    (`x1` and `x2`) in ascending order, with ties averaged.
+
+    * The within-group rank for each observation is determined
+    by ranking the data points within each sample separately,
+
+    * The placement of each observation is calculated by
+    taking the difference between the overall rank and the
+    within-group rank of the observation. Placements can be
+    thought of as measuress of the degree of overlap or
+    separation between two samples.
+    """
+    n_1 = len(x1)
+    n_2 = len(x2)
+
+    # Overall ranks for each obs among combined sample
+    overall_ranks_pooled = rankdata(
+        np.r_[x1, x2], method="average"
+    )
+    overall_ranks_1 = overall_ranks_pooled[:n_1]
+    overall_ranks_2 = overall_ranks_pooled[n_1:]
+    # Within group ranks for each obs
+    within_group_ranks_1 = rankdata(x1, method="average")
+    within_group_ranks_2 = rankdata(x2, method="average")
+
+    placements_1 = overall_ranks_1 - within_group_ranks_1
+    placements_2 = overall_ranks_2 - within_group_ranks_2
+
+    return Holder(
+        n_1=n_1,
+        n_2=n_2,
+        overall_ranks_pooled=overall_ranks_pooled,
+        overall_ranks_1=overall_ranks_1,
+        overall_ranks_2=overall_ranks_2,
+        within_group_ranks_1=within_group_ranks_1,
+        within_group_ranks_2=within_group_ranks_2,
+        placements_1=placements_1,
+        placements_2=placements_2,
+    )
+
+
+def rank_compare_sample_size(
+    reference_sample,
+    synthetic_sample,
+    alpha,
+    power,
+    prop_reference=0.5,
+    alternative="two-sided",
+) -> Holder:
+    """
+    Compute the required sample size for the non-parametric
+    Mann-Whitney U test.
+
+    Parameters
+    ----------
+    reference_sample : array_like
+        Advance information for the reference group.
+    synthetic_sample : array_like
+        Generated `synthetic` data representing the treatment
+        group under the research hypothesis.
+    alpha : float
+        The type I error rate for the test (two-sided).
+    power : float
+        The desired power of the test.
+    prop_reference : float, optional
```

Comment:
still confusing and not consistent with other power functions

use "nobs_ratio"  in signature
and compute and use prop_reference internally so that computations don't need to change

Reply:
This is done.